src/Migrator.EF6.Tools/ProjectReader.cs

```diff
@@ -123,55 +130,17 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
 			return project;
 		}
 
-		public static NuGetFramework GetTargetFramework(string framework)
+		public static NuGetFramework GetTargetFramework(string frameworkString)
 		{
-			switch (framework)
+			bool flag = false;
+			var targetFrameworkMonikerPrefix = _targetFrameworkMonikerPrefixes.FirstOrDefault(t => frameworkString.StartsWith(t.Key, StringComparison.OrdinalIgnoreCase) && (flag = true));
```

Comment:
KeyValuePair is non nullable. Therefore, it is not possible to check for "targetFrameworkMonikerPrefix == null". It would be necessary to use something like that: "string.IsNullOrEmpty(targetFrameworkMonikerPrefix.Key)". But that's not cleaner.

It would be possible to check with Any() and just select an element if the result is true. Another option would be an extension method.

More about the topic [https://stackoverflow.com/questions/5424968/dictionary-firstordefault-how-to-determine-if-a-result-was-found](url)

It's up to you which way you prefer. I don't have any preference.